dask/dataframe/io/hdf.py

```diff
@@ -373,7 +370,7 @@ def read_hdf(pattern, key, start=0, stop=None, columns=None,
     >>> dd.read_hdf('myfile.1.hdf5', '/*')  # doctest: +SKIP
     """
     if lock is True:
-        lock = Lock()
+        lock = get_scheduler_lock()
```

Comment:
Note I added another usage of the new `get_scheduler_lock`. It doesn't break tests but I'd like someone to double-check me here.